render.go

```diff
@@ -83,7 +95,14 @@ func (r *Renderer) enter(w util.BufWriter, n *Node) error {
 	}
 
 	r.hasDest.Store(n, struct{}{})
-	_, _ = w.WriteString(`<a href="`)
+	_, _ = w.WriteString(`<a `)
+	for _, attr := range r.Attributes {
+		_, _ = w.WriteString(attr.Attr)
+		_, _ = w.WriteString(`="`)
+		_, _ = w.WriteString(attr.Value)
```

Comment:
We can use goldmark's [util.EscapeHTML](https://pkg.go.dev/github.com/yuin/goldmark/util#EscapeHTML) or Go std's [template.HTMLEscape](https://pkg.go.dev/html/template#HTMLEscape).
Attribute keys don't need to be escaped but values need to be.